esp32/mods/modhttp.c

```diff
@@ -0,0 +1,1475 @@
+typedef struct mod_http_resource_obj_s {
+	mp_obj_base_t base;
+	struct mod_http_resource_obj_s* next;
+	const char* uri;
+	uint8_t* value;
+	uint16_t value_len;
+	uint8_t mediatype;
+}mod_http_resource_obj_t;
+
+typedef struct mod_http_server_obj_s {
+	httpd_handle_t server;
+	mod_http_resource_obj_t* resources;
+	int max_resources;
+	int num_of_resources;
+}mod_http_server_obj_t;
+
+// From esp_httpd_priv.h
+#define HTTPD_SCRATCH_BUF  MAX(HTTPD_MAX_REQ_HDR_LEN, HTTPD_MAX_URI_LEN)
+
+struct httpd_req_aux {
+    struct sock_db *sd;                             /*!< Pointer to socket database */
+    char            scratch[HTTPD_SCRATCH_BUF + 1]; /*!< Temporary buffer for our operations (1 byte extra for null termination) */
+    size_t          remaining_len;                  /*!< Amount of data remaining to be fetched */
+    char           *status;                         /*!< HTTP response's status code */
+    char           *content_type;                   /*!< HTTP response's content type */
+    bool            first_chunk_sent;               /*!< Used to indicate if first chunk sent */
+    unsigned        req_hdrs_count;                 /*!< Count of total headers in request packet */
+    unsigned        resp_hdrs_count;                /*!< Count of additional headers in response packet */
+    struct resp_hdr {
+        const char *field;
+        const char *value;
+    } *resp_hdrs;                                   /*!< Additional headers in response packet */
+    struct http_parser_url url_parse_res;           /*!< URL parsing result, used for retrieving URL elements */
+};
+
+/******************************************************************************
+ DECLARE PRIVATE FUNCTIONS
+ ******************************************************************************/
+STATIC mod_http_resource_obj_t* find_resource(const char* uri);
+STATIC mod_http_resource_obj_t* add_resource(const char* uri, mp_obj_t value, mp_int_t mediatype);
+STATIC void remove_resource(const char* uri);
+STATIC void resource_update_value(mod_http_resource_obj_t* resource, mp_obj_t new_value);
+STATIC esp_err_t mod_http_resource_callback_helper(mod_http_resource_obj_t* resource , httpd_method_t method, mp_obj_t callback);
+STATIC int mod_http_server_get_mediatype_id(const char* mediatype);
+STATIC bool mod_http_server_get_acceptance(const char* accept_field, uint8_t mediatype_id);
+
+STATIC void mod_http_server_callback_handler(void *arg_in);
+STATIC esp_err_t mod_http_server_callback(httpd_req_t *r);
+
+STATIC void mod_http_client_callback_handler(void *arg_in);
+//STATIC esp_err_t mod_http_client_callback(httpd_req_t *r);
+
+/******************************************************************************
+ DEFINE PRIVATE VARIABLES
+ ******************************************************************************/
+
+STATIC const char* mod_http_mediatype[] = {
+		HTTPD_TYPE_TEXT,
+		MOD_HTTP_MEDIA_TYPE_TEXT_XML,
```

Comment:
This looks inconsistent, that some macro starts with MOD some with HTTPD. Please use the esp-idf macros if possible in all location.